rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+
+```tsx
+export const Accordion: React.FunctionComponent<AccordionProps & React.RefAttributes<HTMLElement>> = React.forwardRef<
+  HTMLElement,
+  AccordionProps
+>((props, ref) =>
+```
+
+```tsx
+export const Avatar = React.forwardRef((props: AvatarProps, ref: React.Ref<HTMLElement>) =>
+```
+
+## Props declared as interfaces vs types
+
+> Some props are defined as interfaces and extended. Others are defined as types and intersected. This isn't always consistent within the same component.
+
+Note: I won't list out every component here. We can scrub once we decide the right approach.
+
+```ts
+export interface AccordionProps extends ComponentProps<AccordionSlots>, Partial<AccordionCommons>
+
+export type AccordionSlots
+export interface AccordionCommons
+```
+
+```ts
+export interface AccordionHeaderProps extends ComponentProps<AccordionHeaderSlots>, Partial<AccordionHeaderCommons>
+
+export type AccordionHeaderSlots
+export interface AccordionHeaderCommons
+```
+
+## Props extending ComponentProps vs. ComponentPropsCompat
+
+> Some props are still extended from ComponentPropsCompat
+
+AvatarProps, BadgeProps, ButtonProps, ...
+
+> Some props have properties using ShorthandPropsCompat
+
+AvatarProps (badge, image, label, icon), BadgeProps (icon), ButtonProps (icon), ...
+
+## Slots and HTML attributes
+
+> Only Accordian and Menu components use ComponentProps<> with slots currently.
+
+AccordionSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+AccordionHeaderSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  button: ARIAButtonShorthandProps;
+  expandIcon: AccordionHeaderExpandIconProps;
+  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  children: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+AccordianItemSlots, AccordianPanelSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuSlots
+
+```ts
+(empty type)
+```
+
+MenuDividerSlots, MenuGroupSlots, MenuGroupHeaderSlots, MenuListSlots, MenuPopoverSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuItemSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+icon?: React.HTMLAttributes<HTMLElement>;
+checkmark?: React.HTMLAttributes<HTMLElement>;
+submenuIndicator?: React.HTMLAttributes<HTMLElement>;
+content: React.HTMLAttributes<HTMLElement>;
+secondaryContent?: React.HTMLAttributes<HTMLElement>;
+```
+
+## Compat and HTML attributes
+
+> The components not yet using slots extend ComponentPropsCompat and provide HTML attributes by extending the associated React HTML attribute types (e.g `React.HTMLAttributes<HTMLElement>`).
+> PortalProps and PopoverTriggerProps do not extend any HTML attributes, but provide children as a property directly in their types.
```

Comment:
`Portal` and `PopoverTrigger` do not render HTML elements, not extending HTML attributes is expected